### ISSUE TITLE
Path metavars inside macro expansions can't be used to refer to macros in some context

### ISSUE THREAD

**Issue by bjorn3** [rust-lang/rust#150019]
I tried this code:

```rust
macro_rules! a {
    ($a:path) => {
        const _: () = $a!();
    };
}

a! {std::println}
```

I expected to see this happen: An error that `println!()` is not allowed in const contexts.

Instead, this happened: ``expected one of `.`, `;`, `?`, `where`, or an operator, found `!` ``

However the following does correctly parse:

```rust
macro_rules! a {
    ($a:path) => {
        const _: () = {
            $a!();
        };
    };
}

a! {std::println}
```

### Meta

`rustc --version --verbose`:
```
rustc 1.92.0 (ded5c06cf 2025-12-08)
binary: rustc
commit-hash: ded5c06cf21d2b93bffd5d884aa6e96934ee4234
commit-date: 2025-12-08
host: x86_64-unknown-linux-gnu
release: 1.92.0
LLVM version: 21.1.3
```

**Comment by fmease** [comment]
Closing as duplicate of https://github.com/rust-lang/rust/issues/143221 from 2025-06-30.
